src/charm.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
+# Copyright 2024 dima.tisnek@canonical.com
+# See LICENSE file for licensing details.
+"""Charmed Gubernator."""
+
+import logging
+
+import ops
+
+from charms.traefik_k8s.v2.ingress import IngressPerAppRequirer
+
+# Log messages can be retrieved using juju debug-log
+logger = logging.getLogger(__name__)
+
+VALID_LOG_LEVELS = ["info", "debug", "warning", "error", "critical"]
+
+
+class HexanatorCharm(ops.CharmBase):
+    """Charm the service."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.ingress = IngressPerAppRequirer(self, port=80, strip_prefix=True)
+        self.framework.observe(self.on['gubernator'].pebble_ready, self._on_gubernator_pebble_ready)
+
+    def _on_gubernator_pebble_ready(self, event: ops.PebbleReadyEvent):
```

Comment:
Nit: because this is all about Gubernator and there's only one container/service, I'd recommend leaving out the `gubernator` from the name:

```suggestion
    def _on_pebble_ready(self, event: ops.PebbleReadyEvent):
```

Reply:
fixed, thanks